libbeat/docs/upgrading-to-7.asciidoc

```diff
@@ -0,0 +1,132 @@
+[[upgrading]]
+== Upgrading
+
+This section gives general recommendations for upgrading Beats shippers:
+
+* <<upgrading-minor-versions>>
+* <<upgrading-6-to-7>>
+* <<upgrade-mapping-template>>
+
+If you're upgrading other products in the stack, also read the
+{stack-ref}/index.html[Elastic Stack Installation and Upgrade Guide]. 
+
+[[upgrading-minor-versions]]
+=== Upgrade between minor versions
+
+As a general rule, upgrading between minor versions (e.g. 7.x to 7.y, where x <
+y) can be done by simply installing the new release and restarting the Beat
+process, because the Beats keep backwards compatibility when it comes to
+configuration and exported fields. Please review the <<release-notes,release
+notes>> for potential exceptions.
+
+Upgrading between non-consecutive major versions (e.g. 5.x to 7.x) is not
+supported. If you are on version 1.x of the Beats, we recommend first upgrading
+to ?.?.
+
+// REVIEWERS: Any specific recommendations for upgrading to a specific version
+// before upgrading to 7.x? I'm not sure we want to even talk about 1.x at this
+// point because it's been EOL for so long. 
+
+[[upgrading-6-to-7]]
+=== Upgrade from 6.x to 7.x
+
+Before upgrading your Beats, review the <<breaking-changes, breaking changes>>
+doc.
+
+If you are planning an upgrade of the full stack ({es}, {kib}, or
+{ls}, in addition to Beats), read the
+{stack-ref}/index.html[Elastic Stack Installation and Upgrade Guide].
+
+We recommend that you fully upgrade {es} and {kib} to version 7.0
+before upgrading Beats.
+
+// REVIEWERS: Is there a min 6.x version that users need to upgrade to *before*
+// upgrading to 7.0?
+//
+// If so, I'll add this:
+//
+// If you are on a Beats 6.x version lower than 6.?, 
+// follow the <<upgrading-to-6.?>> section *before* doing the {es} upgrade.
+
+
+// REVIEWERS: What is the minimum version of 6.x that still runs with 7.0 stack?
+
+If you're not ready to upgrade {es} and {kib} to 7.0, that's alright.
+Beats version 7.0 works with {es} and {kib} version 6.?, so you can
+upgrade Beats now and the rest of the stack later.
+
+[[upgrading-to-6.?]]
+==== Upgrade to 6.?
+
+// REVIEWERS: Do users need to upgrade to a specific version of Beats before
+// upgrading to 7.0? If so, why? I'm assuming the ECS field migration will have
```

Comment:
users must upgrade to 6.7 because of the `_doc` requirement in the template for 7.0.